packages/core-mobile/app/new/features/stake/hooks/useAddStake.ts

```diff
@@ -14,32 +21,42 @@ export const useAddStake = (): {
   const [canAddStake, setCanAddStake] = useState(false)
   const { minStakeAmount } = useStakingParams()
   const { navigateToSwap } = useNavigateToSwap()
+  const { navigateToBuyAvax } = useBuy()
+  const { data: cryptoCurrencies } = useSearchCryptoCurrencies({
+    categories: [ServiceProviderCategories.CryptoOnramp]
+  })
 
-  const handleBuy = useCallback((): void => {
-    // @ts-ignore TODO: make routes typesafe
-    navigate({ pathname: '/buy' })
-  }, [navigate])
+  const cryptoCurrency = useMemo(
+    () =>
+      cryptoCurrencies?.find(
+        crypto => crypto.currencyCode === MELD_CURRENCY_CODES.AVAXC
+      ),
+    [cryptoCurrencies]
+  )
```

Comment:
this logic shouldn't live here. maybe you can make navigateToBuyAvax undefined if we can't find avax c?